packages/hoc/src/asWhitespace/index.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import glamorous from 'glamorous';
+import { BLUE } from 'govuk-colours';
+import {
+  RESPONSIVE_0,
+  RESPONSIVE_1,
+  RESPONSIVE_2,
+  RESPONSIVE_3,
+  RESPONSIVE_4,
+  RESPONSIVE_5,
+  RESPONSIVE_6,
+  RESPONSIVE_7,
+  RESPONSIVE_8,
+  RESPONSIVE_9,
+  MEDIA_QUERIES,
+} from '@govuk-react/constants';
+
+const spacingMap = [
+  RESPONSIVE_0,
+  RESPONSIVE_1,
+  RESPONSIVE_2,
+  RESPONSIVE_3,
+  RESPONSIVE_4,
+  RESPONSIVE_5,
+  RESPONSIVE_6,
+  RESPONSIVE_7,
+  RESPONSIVE_8,
+  RESPONSIVE_9,
+];
+
+const asWhitespace = (Component, marginBottom) => {
```

Comment:
We may add other config options in future so second argument should be a destructured config object `{ marginBottom }` (however see follow up point)